### ISSUE TITLE
FATAL EXCEPTION: IntentService[OdooRegisterService]

### ISSUE THREAD

**Issue by talazoft** [Odoo-mobile/framework#151]
Hello devs, I got error when syncing

FATAL EXCEPTION: IntentService[OdooRegisterService]
Process: com.rd, PID: 20109
java.lang.NullPointerException: Attempt to invoke virtual method 'int odoo.helper.OdooVersion.getVersionNumber()' on a null object reference
                                                                     at odoo.wrapper.OdooWrapper.bindOdooSession(OdooWrapper.java:547)
                                                                       at odoo.wrapper.OdooWrapper.authenticate(OdooWrapper.java:449)
                                                                       at odoo.wrapper.OdooWrapper.authenticate(OdooWrapper.java:440)
                                                                       at com.om.OdooRegisterService.onHandleIntent(OdooRegisterService.java:51)
                                                                       at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:66)
                                                                       at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                       at android.os.Looper.loop(Looper.java:224)
                                                                       at android.os.HandlerThread.run(HandlerThread.java:61)

Im using Odoo RPC v3, with odoo version 7

any help will be very appreciated :)

**Comment by talazoft** [comment]
I have override getOdooVersion on each models but sometimes it still give me forced close with the above error when syncing data

please help

**Comment by talazoft** [comment]
>     private void bindOdooSession(OdooResult response) {
        if(this.odooSession == null) {
            this.odooSession = new OdooSession();
        }
        if(this.mVersion.getVersionNumber() > 7) {  <----- the error was here
            this.odooSession.setCompanyId(response.getInt("company_id"));
        }
        this.odooSession.setDb(response.getString("db"));
        this.odooSession.setSessionId(response.getString("session_id"));
        this.odooSession.setUid(response.getInt("uid"));
        this.odooSession.setUserContext(response.getMap("user_context"));
        this.odooSession.setUsername(response.getString("username"));
    }

is there is any function that can override mVersion number?

**Comment by 74n3r** [comment]
also taking this errors @talazoft  could u solve ?

**Comment by talazoft** [comment]
Well, its not fully solved yet, because sometimes, I'm still experiencing crashes because of this error

Anyway.....
I found that my data is around 2500+ rows when sync started. But my app needs all of that data. So what I did is:
1. Modifiying some tables on server side, reducing usage of M2M relation this time
2. Using adapter.syncDataLimit(limitNumber) for models that potentially returns many rows
